nncf/quantization/precision_init/manual_init.py

```diff
@@ -39,23 +38,31 @@ class ManualPrecisionInitializer(BasePrecisionInitializer):
     def __init__(self,
                  algo: 'ExperimentalQuantizationController',
                  params: ManualPrecisionInitParams,
-                 init_args: QuantizationPrecisionInitArgs = None,
                  hw_precision_constraints: HardwareQuantizationConstraints = None):
-        super().__init__(algo, params, init_args)
+        super().__init__(algo, params, hw_precision_constraints)
         self._bitwidth_per_scope = params.bitwidth_per_scope
 
     def apply_init(self) -> SingleConfigQuantizerSetup:
+        quantizer_setup = self._algo.get_quantizer_setup_for_current_state()
         for pair in self._bitwidth_per_scope:
-            if len(pair) != 2:
-                raise ValueError('Invalid format of bitwidth per scope: [int, str] is expected')
-            bitwidth = pair[0]
-            scope_name = pair[1]
+            bitwidth, scope_name = pair
             is_matched = False
-            for scope, quantizer in self._all_quantizers_per_scope.items():
-                if in_scope_list(str(scope), scope_name):
-                    quantizer.num_bits = bitwidth
+            msg = 'Failed to assign bitwidth={} to `{}`,\n' \
+                  'because it is incompatible with supported quantization configs: {}'
```

Comment:
Can you also add in the message something like "for the specified target hardware" for more clarity?

Reply:
Done